resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/RetrofitClientFactory.scala

```diff
@@ -44,6 +44,12 @@ private[spark] object RetrofitClientFactoryImpl extends RetrofitClientFactory wi
 
   def createRetrofitClient[T](baseUrl: String, serviceType: Class[T], sslOptions: SSLOptions): T = {
     val dispatcher = new Dispatcher(ThreadUtils.newDaemonCachedThreadPool(s"http-client-$baseUrl"))
+    val okHttpClientBuilder = new OkHttpClient.Builder().dispatcher(dispatcher)
+    val resolvedBaseUrl = if (!baseUrl.endsWith("/")) {
+      s"$baseUrl/"
+    } else {
+      baseUrl
+    }
     val serviceUri = URI.create(baseUrl)
```

Comment:
using baseUrl instead of resolvedBaseUrl for the proxy calculation as it's closer to what the user specified, though arguably this could use resolvedBaseUrl throughout